File: Python/#804_uniqueMorseRepresentations.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        code = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        s = set()
        for word in words:
            st = ""
            for i in word:
                charNum = ord(i)-ord('a')
                st += code[charNum]

            if st not in s:
                print(st)
                s.add(st)
        return len(s)

File: Java/#804_uniqueMorseRepresentations.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] alpha={".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set<String> s = new HashSet<String>();
        for(int i=0;i<words.length;i++){
            String trans="";
            for(int j=0;j<words[i].length();j++){
                int t = (int)words[i].charAt(j)-97;
                trans+=alpha[t];
            }
            s.add(trans);
        }
        return s.size();
    }
}
